src/Terminal.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Terminal (terminalGame) where
+
+import Control.Applicative (liftA2)
+import Control.Monad ((<=<))
+import Data.Either.Extra (maybeToEither)
+import Data.Function (on)
+import Data.Map.Strict (empty, findWithDefault)
+import Data.Text qualified as Text
+import Data.Text.IO qualified as TextIO
+import GameLogic (Board (..), Coordinates, Player (..), nextTurn, playAtSpot, winner)
+import System.Console.ANSI (Color (..), ColorIntensity (..), ConsoleLayer (..), SGR (..), clearScreen, setCursorPosition, setSGR)
+import Text.Read (readMaybe)
+
+showPlayerT :: Player -> Text.Text
+showPlayerT = Text.pack . show
+
+showBoard :: Board -> Text.Text
+showBoard (Board board) =
+  "  0   1   2 \n"
+    <> Text.intercalate "\n ---+---+---\n" rows
+  where
+    rows = rowString <$> (zip [0 ..] $ replicate 3 [0 .. 2])
```

Comment:
This `zip` is redundant. Use `zipWith`.

Reply:
`zipWith` what?

Reply:
Curried `rowString`, so it shouldn't be uncurried in the first place.